### PR TITLE
V2 Handshake & enums

### PR DESCRIPTION
**Pull request by a-hansen**
@d-shapiro

### CODE REVIEW

File: modules/core/dslink-core/src/main/java/org/iot/dsa/node/DSEnum.java [hunk 1]
```diff
@@ -1,144 +1,135 @@
...
-    }
-
-    @Override
-    public DSIValue copy() {
+    public DSEnum copy() {
         return this;
     }
 
     @Override
     public DSISynthetic decode(DSElement arg) {
```
Comment (d-shapiro) [inline modules/core/dslink-core/src/main/java/org/iot/dsa/node/DSEnum.java]: should this be static?

Comment (d-shapiro) [inline modules/core/dslink-core/src/main/java/org/iot/dsa/node/DSEnum.java]: should this be static?

Comment (d-shapiro) [inline modules/core/dslink-core/src/main/java/org/iot/dsa/node/DSEnum.java]: decode, that is

Comment (a-hansen) [inline modules/core/dslink-core/src/main/java/org/iot/dsa/node/DSEnum.java]: I wish, but I don't want to use reflection to find the decode method, and you can't put static methods in interfaces.  Instead, we'll use the null instance for decoding.  We'll be persisting the class names, then when deserializing, Class.forName will register the null instance, and I be able to use that for decoding.

Comment (d-shapiro) [inline modules/core/dslink-core/src/main/java/org/iot/dsa/node/DSEnum.java]: lgtm, then

**Pull request merged**